### ISSUE TITLE
Keep ast's wildcard expanded after building LogicalPlan 

### ISSUE THREAD

**Issue by AndrewDi** [pingcap/tidb#8932]
## Feature Request

**Is your feature request related to a problem? Please describe:**
In the implement view feature, SQL from view's definition need to be expanded if sql contains any wildcard, 

**Describe the feature you'd like:**
Because table column info mostly can be obtained during build logical plan，**so I think how about we keep ast's wildcard expanded after logical plan building complete?** 
And current behavior is to reset ast's fields to original ast's fileds after building logical plan, following is the code reference:
https://github.com/pingcap/tidb/blob/6a9ad3fe25e8c835da2dfc83afd458748c92c9ac/planner/core/logical_plan_builder.go#L1806
https://github.com/pingcap/tidb/blob/6a9ad3fe25e8c835da2dfc83afd458748c92c9ac/planner/core/logical_plan_builder.go#L1907

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by AndrewDi** [comment]
@XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
The code is introduced in https://github.com/pingcap/tidb/pull/2767.
Please check whether there will be negative effects if we do the modification.

**Comment by AndrewDi** [comment]
@XuHuaiyu Do we have any test case to reproduce this problem, Following test case from PR #2767 runs fine.
```mysql
create table multiexec (a int, b int);
insert multiexec values (1, 1), (2, 2);
prepare stmt from 'select a from multiexec where b = ? order by b';
set @val=1;
execute stmt using @val;
set @val=2;
execute stmt using @val;
deallocate prepare stmt;
drop table multiexec;
```

**Comment by XuHuaiyu** [comment]
Have you checked the result of `make dev`?

**Comment by XuHuaiyu** [comment]
Yep, the panic mentioned in that issue may not happen, since we've removed the corresponded code line.

I think it's ok to make this modification.

PTAL @winoros @zz-jason

**Comment by winoros** [comment]
It changed the AST tree itself.
When will the `ast.Node.Restore` be called? I'm not sure whether this will break its functionality.

**Comment by AndrewDi** [comment]
@winoros I may call `ast.Node.Restore` when `Create View`.

**Comment by XuHuaiyu** [comment]
@winoros It seems that we do not re-use the AST now.

**Comment by XuHuaiyu** [comment]
@AndrewDi I wonder whether we need this to help us to perform the P1 tasks mentioned in https://github.com/pingcap/tidb/blob/master/docs/design/2018-10-24-view-support.md?
This proposal only considers the **SIMPLEST* functions of view. We need another detailed proposal and discussion on other functions.

**Comment by AndrewDi** [comment]
IMO if I want to get a `Expanded AST Tree`, there are two ways:
1. Modify the origin one ---If we do not reuse origin ast tree, this way is more simple. 
2. Return a new one

**Comment by XuHuaiyu** [comment]
@winoros TiDB changes part of the AST now, like [here](https://github.com/pingcap/tidb/blob/master/planner/core/logical_plan_builder.go#L1193).

**Comment by XuHuaiyu** [comment]
@AndrewDi You're right, but detailed investigation needs to be done before coding. Since no one really ensures which way is better. And advanced functions of view are complicated.

**Comment by AndrewDi** [comment]
@XuHuaiyu Yes, I just need modify `SelectStmt.Fields.Fields`, so I think it's not a hard decision.

**Comment by AndrewDi** [comment]
@XuHuaiyu @winoros Restore `SelectStmt` to SQL func completed, how about discuss this issue again? [view/keep_ast_expand ](https://github.com/AndrewDi/tidb/tree/view/keep_ast_expand)
